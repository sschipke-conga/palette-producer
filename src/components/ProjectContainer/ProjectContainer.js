import './ProjectContainer.scss';
import React from 'react';
import ProjectCard from '../ProjectCard/ProjectCard';
import PropTypes from 'prop-types';
import { IoIosAddCircleOutline } from 'react-icons/io';

export const ProjectContainer = ({ select, projects, palettes, removePalette, removeProject }) => {
  let displayProjects = null;
  if (projects.length !== 0 && palettes.length !== 0) {
    displayProjects = projects.map((project, index) => {
      return (
        <ProjectCard
          key={'ProjectCard' + index}
          project={project}
          palettes={palettes[index]}
          select={select}
          removePalette={removePalette}
          removeProject={removeProject}
        />
      )
    })
  }

  return (
    <div className='ProjectContainer'>
      {projects.length === 0 && (<>
      <h4 className='no-projects'>Get started by your first palette to your first project!</h4>
      <p className='explanation'>Once you've made a palette, double click it to select it.</p>
      </>)}
      {projects.length !== 0 && <div className='add-project-container' onClick={() => select({name: ''}, {name: ''})}>
        <IoIosAddCircleOutline className='add-project-icon' />
      </div>}
      {displayProjects}
    </div >
  )
}

export default ProjectContainer;

ProjectContainer.propTypes = {

}
