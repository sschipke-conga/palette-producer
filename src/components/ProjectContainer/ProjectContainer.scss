div .ProjectContainer {
    top: 50%;
    height: 90vh;
    width: 100vw;
    transition: 0.3s ease-in-out;
    background: rgb(255, 243, 191);
    margin: 20px auto;
    padding-top: 10px;
    overflow: hidden;
    min-width: 90vw;
button {
    border: 1px solid purple;
    border-radius: 5px;
    height: 60px;
    font-size: 1.5em;
    width: 20%;
    margin: 20px auto;
    background: #FDF7DB;
    &:hover {
      color: white;
      background: purple;
    }
  }
}

.add-project-container {
  align-items: center;
  background: rgb(244, 244, 244);
  cursor: pointer;
  display: flex;
  height: 100px;
  justify-content: center;
  margin: 10px auto;
  &:hover {
    background: rgb(194, 194, 194);
  } 
  &:active {
    background: rgb(177, 177, 177);
  }
}

.add-project-icon {
  height: 50px;
  margin: 10px;
  width: 50px;
}

.no-projects {
  color: white;
  font-size: 2em;
  margin: 10px auto;
}

.explanation {
  color: white;
  font-size: 1.5em;
}