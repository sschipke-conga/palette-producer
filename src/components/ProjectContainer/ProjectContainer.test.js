import React from "react";
import { ProjectContainer, mapStateToProps, mapDispatchToProps} from "./ProjectContainer";
import { shallow } from "enzyme";

describe('ProjectContainer', () => {
  const mockSelect = jest.fn()
    const mockProjects = [{
      name: "Project"
    }];
    const mockPalettes = [{
      id: 1,
      name: "Palette",
      project_id: 1,
      color1: "#FFFFFF",
      color2: "#FFFFFF",
      color3: "#000000",
      color4: "#FFFFFF",
      color5: "#FFFFFF"
    }];
    let wrapper;
  beforeEach(() => {
      wrapper = shallow(
        <ProjectContainer
          allProjects={mockProjects}
          allPalettes={mockPalettes}
        />
      );
    });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
    describe("select", () => {
      describe('alt snapshot', () => {
          const mockSelect = jest.fn();
          const mockProjects = [
            {
              name: "Project"
            }
          ];
          let wrapper;
          beforeEach(() => {
            wrapper = shallow(
              <ProjectContainer
                allProjects={mockProjects}
                allPalettes={[]}
              />
            );
          });
          it('should match the snapshot', () => {
            expect(wrapper).toMatchSnapshot()
          })
      })
    });
  describe('mapStateToProps', () => {
    it('mapStateToProps the user and the state of the menu', () => {
      const mockUser = { username: 'Dave', id: 4 }
      const mockState = {
        allPalettes: mockPalettes,
        allProjects: mockProjects
      };
      const expected = {
        allPalettes: mockPalettes,
        allProjects: mockProjects
      };
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  })
})
