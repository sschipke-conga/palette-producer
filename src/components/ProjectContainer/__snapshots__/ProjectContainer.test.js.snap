// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer select alt snapshot should match the snapshot 1`] = `
<div
  className="ProjectContainer modal-div"
/>
`;

exports[`ProjectContainer should match the snapshot 1`] = `
<div
  className="ProjectContainer modal-div"
>
  <Connect(ProjectCard)
    key="ProjectCard0"
    palettes={Array []}
    project={
      Object {
        "name": "Project",
      }
    }
  />
</div>
`;
